doc/changes/DM-32682.feature.rst

```diff
@@ -0,0 +1,3 @@
+`lsst.pipe.base.TaskMetadata` can now be used as metadata for `~lsst.pipe.base.Task`.
```

Comment:
This depends entirely on how you've set up butler. At the moment butler will give you back what you put in -- we don't have any code for casting the result of a butler.get to a new type. The plan was to try to see if the butler can cast a PropertySet to a TaskMetadata on get but I don't think that affects this ticket. Making butler handle older metadata is a different ticket.

Reply:
Also, since the `*_metadata` dataset type is defined as a `PropertySet` you can't even run a pipeline with `TaskMetadata` as the storage class without first changing the dataset type definition.